### ISSUE TITLE
my own dataset

### ISSUE THREAD

**Issue by Kangzf1996** [zhulf0804/PREDATOR#1]
If I want to train the data on my own dataset, how should I do?
Thank you!

**Comment by zhulf0804** [comment]
Write your dataset similar as data/ThreeDMatch.py#L11, and new your_dataset.yaml file similar as configs/threedmatch.yaml. 

Then maybe you are ready to train with minor modifications.

Good luck.

**Comment by Kangzf1996** [comment]
> Write your dataset similar as data/ThreeDMatch.py#L11, and new your_dataset.yaml file similar as configs/threedmatch.yaml.
> 
> Then maybe you are ready to train with minor modifications.
> 
> Good luck.

Thank you for your reply.
I'm wondering what's the info.pkl file, it's generated by the ThreeDMatch.py?

**Comment by zhulf0804** [comment]
> > Write your dataset similar as data/ThreeDMatch.py#L11, and new your_dataset.yaml file similar as configs/threedmatch.yaml.
> > Then maybe you are ready to train with minor modifications.
> > Good luck.
> 
> Thank you for your reply. I'm wondering what's the info.pkl file, it's generated by the ThreeDMatch.py?

No, it's a file containing 3DMatch information generated before, and you don't need to care about it for your own dataset.
In fact, you only need to think about how to return dict similar as data/ThreeDMatch.py#L70 in your __getitem__() function for your dataset.

**Comment by Kangzf1996** [comment]
> > > Write your dataset similar as data/ThreeDMatch.py#L11, and new your_dataset.yaml file similar as configs/threedmatch.yaml.
> > > Then maybe you are ready to train with minor modifications.
> > > Good luck.
> > 
> > 
> > Thank you for your reply. I'm wondering what's the info.pkl file, it's generated by the ThreeDMatch.py?
> 
> No, it's a file containing 3DMatch information generated before, and you don't need to care about it for your own dataset. In fact, you only need to think about how to return dict similar as data/ThreeDMatch.py#L70 in your **getitem**() function for your dataset.

I noticed that the dict has been generated by the info.pkl, could I generate a .pkl file like 3Dmatch and then return dict similar as data/ThreeDMatch.py#L70. I'm not sure whether this method is ok.

**Comment by zhulf0804** [comment]
> > > > Write your dataset similar as data/ThreeDMatch.py#L11, and new your_dataset.yaml file similar as configs/threedmatch.yaml.
> > > > Then maybe you are ready to train with minor modifications.
> > > > Good luck.
> > > 
> > > 
> > > Thank you for your reply. I'm wondering what's the info.pkl file, it's generated by the ThreeDMatch.py?
> > 
> > 
> > No, it's a file containing 3DMatch information generated before, and you don't need to care about it for your own dataset. In fact, you only need to think about how to return dict similar as data/ThreeDMatch.py#L70 in your **getitem**() function for your dataset.
> 
> I noticed that the dict has been generated by the info.pkl, could I generate a .pkl file like 3Dmatch and then return dict similar as data/ThreeDMatch.py#L70. I'm not sure whether this method is ok.

It's ok. But I think it is a suboptimal option.

**Comment by Kangzf1996** [comment]
> > > > > Write your dataset similar as data/ThreeDMatch.py#L11, and new your_dataset.yaml file similar as configs/threedmatch.yaml.
> > > > > Then maybe you are ready to train with minor modifications.
> > > > > Good luck.
> > > > 
> > > > 
> > > > Thank you for your reply. I'm wondering what's the info.pkl file, it's generated by the ThreeDMatch.py?
> > > 
> > > 
> > > No, it's a file containing 3DMatch information generated before, and you don't need to care about it for your own dataset. In fact, you only need to think about how to return dict similar as data/ThreeDMatch.py#L70 in your **getitem**() function for your dataset.
> > 
> > 
> > I noticed that the dict has been generated by the info.pkl, could I generate a .pkl file like 3Dmatch and then return dict similar as data/ThreeDMatch.py#L70. I'm not sure whether this method is ok.
> 
> It's ok. But I think it is a suboptimal option.

well, I'll try your method. btw how could I generate my own info.pkl file?

**Comment by Kangzf1996** [comment]
> > > Write your dataset similar as data/ThreeDMatch.py#L11, and new your_dataset.yaml file similar as configs/threedmatch.yaml.
> > > Then maybe you are ready to train with minor modifications.
> > > Good luck.
> > 
> > 
> > Thank you for your reply. I'm wondering what's the info.pkl file, it's generated by the ThreeDMatch.py?
> 
> No, it's a file containing 3DMatch information generated before, and you don't need to care about it for your own dataset. In fact, you only need to think about how to return dict similar as data/ThreeDMatch.py#L70 in your **getitem**() function for your dataset.

I'm not clear about the struct of dict in the getitem(), do you have some detailed information? Thank you!
